src/libs/actions/BankAccounts.js

```diff
@@ -330,7 +330,11 @@ function fetchUserWallet() {
 function fetchFreePlanVerifiedBankAccount(stepToOpen) {
     // We are using set here since we will rely on data from the server (not local data) to populate the VBA flow
     // and determine which step to navigate to.
-    Onyx.set(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: true});
+    // We temporarily keep the achData state to prevent UI changes while fetching.
```

Comment:
What happens if we do not do this? At what point is it getting overwritten?

Reply:
also I think this comment should be placed above the line with `achData`

Reply:
The button text is [updated](https://github.com/Expensify/App/blob/22c872cfb2409cc0e1e2b774102f2da4a0db09f6/src/pages/workspace/WorkspaceCardPage.js#L76-L89) when `achData.state` changes. So the UI changes when we set `achData` without the previous state. It is overwritten by setting `achData.state = lodashGet(achData, 'state', '');` and then merging it [here](https://github.com/Expensify/App/blob/22c872cfb2409cc0e1e2b774102f2da4a0db09f6/src/libs/actions/BankAccounts.js#L70).

Reply:
>It is overwritten by setting achData.state = lodashGet(achData, 'state', '');

Hmm where does that happen? Sorry I'm not seeing it 🙃

Reply:
We are passing `achData` with the new state to [goToWithdrawalAccountSetupStep](https://github.com/Expensify/App/blob/62049474b1a681a5a68b89df6e5a7928b36c7981/src/libs/actions/BankAccounts.js#L484) and then merging it [here](https://github.com/Expensify/App/blob/22c872cfb2409cc0e1e2b774102f2da4a0db09f6/src/libs/actions/BankAccounts.js#L70). Because `...achData` is called after `...newACHData`, we are overriding the temporarily stored value.

Reply:
Just posted a semi-related issue in https://github.com/Expensify/App/pull/4477#issuecomment-895415780, linking for context in case its the same issue

Reply:
@Jag96 Took a quick look, but not sure if they have the same "root cause" besides the local state is reset each time we fetch information about the bank account. One caveat to keep in mind is that we attempted to port this logic over from Web-Secure and retain the functionality we have there - but I'm sure there will be some edge cases to work through as the flows are not exactly the same.